### DISCUSSION TITLE
Anyone have an issue where the new agile rates aren't loading?

### DISCUSSION THREAD

**Discussion by JRascagneres** [BottlecapDave/HomeAssistant-OctopusEnergy#623] [General]
Just noticed that todays rates haven't been pulled down but they're available on third party sites / octopus UI... Anyone seeing similar?

**Comment by DLATHA** [comment]
Having the same issue. This happened last week and a reboot of HA fixed it. May have been just good timing as nothing appears to fix it today. Running a curl command to pull the rates works fine.
Debug error shows:
DEBUG (MainThread) [custom_components.octopus_energy.coordinators.electricity_rates] Failed to retrieve new electricity rates for ### so using cached rates

**Reply by JRascagneres** [reply]
Yeah, I've tried restarting the integration but doesn't help...
@BottlecapDave any ideas?

**Reply by Markyt77** [reply]
Same for me, has happened once or twice recently but restarting, reloading etc today isn't picking up the new agile rates

**Comment by pcb1962** [comment]
Mine are late today, restart of HA hasn't fixed it.

**Reply by stevetrease** [reply]
Same. Late today and a reboot of HA hasn't sorted it.

**Comment by pcb1962** [comment]
Just arrived

**Reply by DLATHA** [reply]
Mine too

**Reply by JRascagneres** [reply]
Not for me yet... Lets wait

**Comment by JRascagneres** [comment]
Ah its picked it up now, just seemed slow, never hit this before. Thanks guys

**Reply by stevetrease** [reply]
Mine just updated. Around about 1800, so two hours late. Strange.

**Comment by BottlecapDave** [comment]
Hello. I'm sorry everyone is seeing this issue. The only thing I can think of is that a design decision I made at the start of the integration is starting to come home to roost. In order to get rate information as quickly as possible, the rates are retrieved on the hour and half hour. With the popularity of the integration, this is probably posing some additional strain on the OE servers, which might be having issues for other reasons which are making the issue worst. It might be other services are also making this worst by bombarding the API with requests.

I have a branch, which I'm in the final stages of testing that should help reduce strain on the OE servers and help reduce any issues caused by the integration. I don't have time to work on the integration tonight, but I'll try and release a beta version within the next couple of days with these changes.

**Reply by JRascagneres** [reply]
No problem, much appreciated. I saw that issue you raised. We really appreciate all the work you do on this integration! (maybe not OE haha)

Out of interest... Whats the new plan to pull data?

**Reply by BottlecapDave** [reply]
A bit of insomnia has resulted in me being able to slice a beta release. The discussion around it can be found at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/626 and the beta can be found at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/v9.2.0-beta1. Full details on the new approach can be found on the discussion

**Reply by Markyt77** [reply]
I've just installed the beta after not getting todays agile rates and it worked first time, brilliant work @BottlecapDave

**Reply by JRascagneres** [reply]
I'll likely pull this down soon 😄 - Was waiting for saving session to ensure there isn't an issue with that.

**Reply by DLATHA** [reply]
Beta worked for me too but the target rates didn't work so reverted back for now until I have time to look why.

**Reply by JRascagneres** [reply]
I just updated now and it didn't load the prices. Anything I can poke? I did get these errors (I'm on agile):
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/3264971/0d6633fa-8993-48d6-a056-f0cb8f3b649c)

**Reply by JRascagneres** [reply]
Will revert for now I think. Hopefully it'll pick up prices soonish

**Reply by JRascagneres** [reply]
I've no idea why it isn't pulling the new prices. Is there anything I can do to force to try to pull?

**Reply by borpin** [reply]
Could you create a service @BottlecapDave to allow users to force a refresh?

**Reply by JRascagneres** [reply]
That'd be great. Could potentially make that even automatically do extra logging to find any issues.

**Reply by JRascagneres** [reply]
I'm hoping this will sort itself at some point tonight so I know the 'current rate' at each time. I can plan charging with external data sources

**Reply by BottlecapDave** [reply]
An additional service will just make the situation worse if the APIs are failing to scale. Also due to the architecture of how data is retrieved within HA the service isn't possible.

As for the raised **warnings**, this is part of the fallback (as per the FAQ) which is raised when the API cannot be connected for some reason and will automatically retry steadily waiting longer. If you're not seeing any other accompanying warnings (e.g. `Octopus Energy server error` or `Failed to send request`), the most likely scenario is timeouts when attempting to connect to the API. These are set to 15 seconds (which should be more than enough for an API call). This could be caused by the API being slow to respond (due to overwhelming demand and scaling issues), internet issues on your side or something else.

I'm currently adding some additional logging around timeouts and raised issues.

Unfortunately these changes alone won't necessarily help until the majority of users are using this version of the integration.

**Reply by JRascagneres** [reply]
I'm not seeing any other errors or warnings so I'm struggling to see what's happening. But yeah might be nice to have additional logging around it. Print HTTP status codes etc

**Comment by JRascagneres** [comment]
Yeah ended up hitting this again. Assume OE have changed something on their end recently thats clashing with the way the integration pulls data as I've never really had this before and now a couple days in a row.

**Reply by pcb1962** [reply]
Got mine at 5pm today, same as yesterday, previously it was always 4pm

**Comment by Markyt77** [comment]
v9.2.0-Beta 1 got things working properly yesterday but today the rates have been up on OE since around 16:00 but again missing from HA as of 16:35.

**Reply by Markyt77** [reply]
Rates arrived at 16:50

**Reply by DLATHA** [reply]
Strange as I put the beta back on today and it actually triggered early at 15:55 for me. Target rates are working for me now too. They don't keep the changes sent via the service update though after a reboot and I'm sure they did before.

**Reply by Markyt77** [reply]
Pretty sure target rate changes are not retained after a restart, think it's mentioned in the docs

**Reply by BottlecapDave** [reply]
Correct, any target rate changes made via the service are currently not persisted between restarts. You can kinda get around this by adding the home assistant start event as a trigger for any automation that calls the service.
